opsgenie/resource_opsgenie_notification_rule_test.go

```diff
@@ -0,0 +1,175 @@
+				return err
+			}
+			for _, r := range resp.SimpleNotificationRules {
+				deleteRequest := &notification.DeleteRuleRequest{
+					UserIdentifier: u.Id,
+					RuleId:         r.Id,
+				}
+				if _, err := client.DeleteRule(context.Background(), deleteRequest); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func TestAccOpsGenieNotificationRule_basic(t *testing.T) {
+	randomName := acctest.RandString(6)
+	config := testAccOpsGenieNotificationRule_basic(randomName)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieNotificationRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieNotificationRuleExists("opsgenie_notification_rule.test", randomName),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieNotificationRuleDestroy(s *terraform.State) error {
+	client, err := notification.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_notification_rule" {
+			continue
+		}
+		req := notification.GetRuleRequest{
+			UserIdentifier: rs.Primary.Attributes["username"],
+			RuleId:         rs.Primary.Attributes["id"],
+		}
+		_, err := client.GetRule(context.Background(), &req)
+		if err != nil {
+			x := err.(*ogClient.ApiError)
+			if x.StatusCode != 404 {
+				return errors.New(fmt.Sprintf("Notification rule still exists: %s", x.Error()))
+			}
+		}
+	}
+
+	return nil
+}
+
+func testCheckOpsGenieNotificationRuleExists(name, username string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		client, err := notification.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		req := notification.GetRuleRequest{
+			RuleId:         rs.Primary.Attributes["id"],
+			UserIdentifier: rs.Primary.Attributes["username"],
+		}
+
+		_, err = client.GetRule(context.Background(), &req)
+		if err != nil {
+			return fmt.Errorf("Notification rule does not exist (and it should).")
+		} else {
+			log.Printf("Notification rule found.")
+		}
+
+		return nil
+	}
+}
+
+func testAccOpsGenieNotificationRule_basic(randomName string) string {
+	return fmt.Sprintf(`
+resource "opsgenie_user" "test" {
+  username  = "genieuser-%s@opsgenie.com"
+  full_name = "Acceptance Test User"
+  role      = "User"
+}
+
+resource "opsgenie_notification_rule" "test" {
+  name = "genierule-%s"
+  username = opsgenie_user.test.username
+  action_type = "schedule-end"
+  notification_time = ["just-before", "15-minutes-ago"]
+  steps {
+    contact {
+      method = "email"
+      to = "genieuser-%s@opsgenie.com"
+    }
+  }
+  order = 0
+  enabled = true
+  repeat {
+    loop_after = 2
+  }
+  timeRestriction {
```

Comment:
should be time_restriction

Reply:
Let me fix it. and run the test.

Reply:
Fixed. tests are passing. resolving the conversation.